src/main/scala/zio/actors/ActorSystem.scala

```diff
@@ -222,8 +222,14 @@ final class ActorSystem private[actors] (
           obj      <- readFromWire(worker)
           envelope = obj.asInstanceOf[Envelope]
           actorMap <- refActorMap.get
-
-          _ <- actorMap.get("/" + envelope.recipient.split("/").last) match {
+          remoteActorPath <- for {
```

Comment:
You can drop the entire for-comprehension with: `resolvePath(envelope.recipient).map(_._4)`